gui/gui/src/TGSpeedo.cxx

```diff
@@ -300,6 +315,17 @@ Bool_t TGSpeedo::HandleButton(Event_t *event)
    return kTRUE;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Change the circular buffer size (used for the automatic mean calculation).
+
+void TGSpeedo::SetBufferSize(Int_t size)
+{
+   fBufferSize = size;
+   fBuffer.clear();
+   fBuffer.reserve(fBufferSize);
```

Comment:
if you did a `resize` here instead of a `reserve` you wouldn't need the if/else afterwards on the condition `if ((Int_t)fBuffer.size() < (fBufferCount + 1))`. You could just always fill element number `fBufferCount`

Reply:
The problem is that when we calculate the mean value, we average over all elements inside fBuffer. If BufferSize = 10, but we have only filled two times the buffer, one would be averaging garbage. Thus, the if/else condition is needed for the initial entries.

Reply:
What the most common operation? Adding or getting the mean?  Which ever is most common is the one you (probably) have to optimize.

Reply:
Every time you add, GetMean gets calculated (DrawNeedle function is called). GetMean might be also called when updating the graphics layout. So I would say GetMean is called more often than adding. Thus the if-else when adding is better than having it in the mean, I guess.

Reply:
@eguiraud that was my first implementation, but after discussion with @ferdymercury I agreed that the mean was then wrongly calculated until the buffer is full (which is harmless most of the time...)